### PR TITLE
fix scatter op extern func name error bug

### PR DESCRIPTION
**Pull request by thisjiang**
As title.

### CODE REVIEW

**Comment by zzk0** [comment]
cinn/runtime/cuda/cuda_instrinsics_float16.cc

也要对应做修改

File: cinn/runtime/cuda/cinn_cuda_runtime_source.cuh [hunk 1]
```diff
@@ -595,7 +595,7 @@ CINN_NVGPU_GT_NUM(int64, long long int)
 #undef CINN_NVGPU_GT_NUM
 
 #define CINN_CUDA_INDEX_ADD(TYPE_SUFFIX, TYPE)                                \
```
Comment (zzk0) [inline cinn/runtime/cuda/cinn_cuda_runtime_source.cuh]: CINN_CUDA_INDEX_ADD

顺便也将宏的名字进行替换

CINN_NVGPU_INDEX_ADD

File: cinn/runtime/cuda/cuda_intrinsics.cc [hunk 2]
```diff
@@ -334,16 +334,16 @@ CINN_REGISTER_HELPER(cuda_intrinsics) {
 
 #undef _REGISTER_CINN_NVGPU_GT_NUM
 
-#define _REGISTER_CINN_CUDA_INDEX_ADD(TYPE_SUFFIX, TYPE)                        \
```
Comment (zzk0) [inline cinn/runtime/cuda/cuda_intrinsics.cc:337]: _REGISTER_CINN_CUDA_INDEX_ADD

---->

_REGISTER_CINN_NVGPU_INDEX_ADD

**Review by Shixiaowei02 — APPROVED** [review]

**Review by SunNy820828449 — APPROVED** [review]
LGTM

**Pull request merged**